osu.Game/Screens/Menu/ConfirmExitDialog.cs

```diff
@@ -37,9 +38,14 @@ private void load(INotificationOverlay notifications)
             {
                 string text = "There are currently some background operations which will be aborted if you continue:\n\n";
 
-                foreach (var n in notifications.OngoingOperations)
+                var ongoingOperations = notifications.OngoingOperations.ToArray();
+
+                foreach (var n in ongoingOperations.Take(Math.Min(ongoingOperations.Length, 10)))
```

Comment:
don't need the `Math.Min()` if using `Take()` already, i believe `Take()` is documented to sanely handle cases where there are less objects in the enumerable than requested (it'll just return everything)